### PR TITLE
Fix OAuth login buttons

### PR DESCRIPTION
**Pull request by timroes**
## What

This fixes several issues with the OAuth connectors atm:

* The new `connectorFormContext` does no longer contain `selectedService` and `selectedConnector`, but the consuming places were not adjusted to that yet. This fixes the consuming places and the types.
* We were no longer calling `getValues` since the refactor to the `useAuthentication` hook, which would cause connectors (like Bing Ads), which had a `predicateKey` referencing a field that only had a `const` value to no longer show. Since the `getValues` method is the one that put that `const` value as a default value into the Formik values.
* Some connectors would say "Re-authenticate" even before authentication (e.g. Bing), because they had a `default` value for one of the OAuth flow fields of empty string, which caused our check for `undefined` though to now think the OAuth flow already ran. This was now changed to a truthy change, so empty string default values in OAuth flow fields no longer mark the button as "Reauthenticate" from the beginning.

### CODE REVIEW

File: airbyte-webapp/src/views/Connector/ConnectorForm/connectorFormContext.tsx [hunk 1]
```diff
@@ -63,7 +63,7 @@ export const ConnectorFormContextProvider: React.FC<React.PropsWithChildren<Conn
 
   const ctx = useMemo<ConnectorFormContext>(() => {
     const unfinishedFlows = widgetsInfo["_common.unfinishedFlows"] ?? {};
-    return {
+    const context: ConnectorFormContext = {
```
Comment (timroes) [inline airbyte-webapp/src/views/Connector/ConnectorForm/connectorFormContext.tsx:66]: :information_source: Assigning this object to a variable that's explicitally types, will cause TypeScript to now properly check if the object has keys that it shouldn't have.

**Review by dizel852 — APPROVED** [review]
Code LGTM
Have not tested locally

**Comment by tealjulia** [comment]
Tested a handful locally and am seeing the OAuth button.

I do see a bug where the auth method is persisting if i navigate away and often no default auth method is being applied if it is in a oneOf.

**Review by edmundito — APPROVED** [review]
Code changes look good. I have not tested this.

**Review by lmossman — APPROVED** [review]
Tested both the OAuth buttons and the empty numeric field behavior locally and it looks to be behaving as expected 👍

**Pull request merged**